source/core/ExecutionContext.cpp

```diff
@@ -267,8 +267,8 @@ void AppendCallChainString(StringRef stringRef)
     if (callChain->Length() > 0) {
         stringRef->AppendStringRef(*callChain->BeginAt(callChain->Length() - 1));
     }
-    stringRef->AppendCStr("\r\n");
     itemType->ClearData(&callChain);
+    free(callChain);
```

Comment:
I'm sorry, I missed the ClearData() call you already had here because a chain of inline comments had been added right before it in the diff review. 

The ClearData is sufficient, you should remove the extra free() (which is probably not causing extra issues because ClearData nulls callChain and free silently ignored null pointers).
Sorry for the confusion.